### DISCUSSION TITLE
How do I access the API methods?

### DISCUSSION THREAD

**Discussion by AgentSamSA** [ndresx/react-countdown#145] [Q&A] [answered]
Hello, I'm trying to write helper functions to set the timer for certain increments of time (15s, 30s, 1min, etc) and then start the timer from there (or start the timer once the timer has been set to a value). How do I utilize the start(), pause(), stop() etc methods?

For reference, this is the base helper function for setting to a time value:
  
```
const addTime = (ms) => {
    const time = Date.now() + ms;
    setEndTimestamp(time);
    console.log(endTimestamp);
  }
```

Here is an example function for setting to fifteen seconds:
 ```
const addFifteen = () => {
      addTime(15000);
      //insert call to start method here//
    }
```

I feel like I'm missing something obvious, as I haven't gotten the methods to work. Any ideas?

**Comment by ndresx** [comment, marked as answer]
Hello, and sorry for my late response. I've updated the API section https://github.com/ndresx/react-countdown#api-reference to include an example. Hopefully, this still helps someone.

**Comment by adexzy1** [comment]
can this be done using functional components? all i want to access is the start function

**Reply by lancebelarmino** [reply]
You can access the API by attaching a `ref` to the component.

```
import React, { useRef, useEffect } from 'react';

const Timer = () => {
  const countdownRef= useRef();

  console.log(countdownRef.current.api);
  
  useEffect(() => {
    countdownRef.current.api.start();
  })

  return (
    <Countdown
      ref={countdownRef}
      date={Date.now() + 5000}
      autoStart={false}
    />
  );
};
```
